ffi/src/lib.rs

```diff
@@ -77,13 +148,30 @@ pub struct TransactionChange {
     pub changes: Box<[Entid]>,
 }
 
+ /// A C representation of the list of changes provided by the transaction observers.
+ /// Provides the list of changes as the length of the list.
+///
+/// #Safety
+///
+/// Callers are responsible for managing the memory for the return value.
+/// A destructor `destroy` is provided for releasing the memory for this
+/// pointer type.
 #[repr(C)]
 #[derive(Debug)]
 pub struct TxChangeList {
     pub reports: Box<[TransactionChange]>,
     pub len: usize,
 }
 
+/// A C representation Rust's Option.
```

Comment:
Nit -- lots of `Option` `Some` `None` quotes.

Is this really the standard way to declare this stuff for FFIs?  That surprises me, because it means every crate will  have its own, subtly different, external `Option` representation.  I thought "the point" of Rust's `Option` representation was that it _transparently_ was null or the pointer as appropriate across the FFI boundary.  Or is it that you're allocating in a way that needs special handling in Rust?